product_default_image/models/product_template.py

```diff
@@ -0,0 +1,299 @@
+        target = company.product_image_target
+
+        if target not in ['category', 'global_category']:
+            return
+
+        for record in self:
+
+            if not record.auto_change_image:
+                continue
+
+            if record.img_type == 'custom':
+                continue
+
+            categ = record.categ_id
+
+            img_args = {
+                'to_type': 'default_category',
+                'in_cache': True,
+            }
+            if not categ.image and target == 'global_category':
+                img_args['to_type'] = 'default_global'
+
+            record.change_template_image(**img_args)
+
+    @api.multi
+    def change_template_image(self, to_type, to_img_bg=None, in_cache=False):
+        """ Change product template image to specified to_type's image
+            or to_img_bg.
+
+            Args:
+                to_type:
+                    Value specified allowed from img_type field which the
+                    image should be targeted to. Accepted values are:
+
+                        'default_global'
+                            * product_image field defined in res.company
+
+                        'default_category',
+                            * product's categ_id field's image
+
+                        'no_image'
+                            * Deletes the image
+
+                        'custom'
+                            * Set to the image you specify using to_img_bg.
+                            Note that targeting to custom will prevent the
+                            image from automatically being changed in the
+                            future until deleted or manually re-mapped to a
+                            default image.
+
+                to_img:
+                    Can be used as an override if the template
+                    images should be changed to a specific image.
+                    Image is resized to image big value and scaled
+                    down to medium and small values.
+
+                    Note that to_type is required even if to_img_bg
+                    arg is supplied.
+
+                in_cache:
+                    Set True if calling this method to change the images
+                    in cache (helpful when calling from an onchange method
+                    in product.template)
+
+            Returns:
+                recordset if tmpl images have been changed.
+
+                None/False if any of below conditions are True.
+
+        """
+        if any([
+                to_type not in [
+                    'default_global', 'default_category',
+                    'no_image', 'custom'],
+                to_type == 'custom' and not to_img_bg,
+                to_type == 'no_image' and to_img_bg,
+                not isinstance(in_cache, bool)]):
+            return
+
+        # {img_bg: records}
+        write_map = defaultdict(lambda: self.env['product.template'].browse())
+
+        if to_img_bg:
```

Comment:
Hmmm I don't like this method at all, but I don't want to make you rewrite it either because I think this may be somewhat of a logic bomb.

@tedsalmon can you please make this more elegant?